pkg/services/librarypanels/models.go

```diff
@@ -0,0 +1,42 @@
+package librarypanels
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// LibraryPanel is the model for library panel definitions.
+type LibraryPanel struct {
+	Id       int64
+	OrgId    int64
+	FolderId int64
+	Title    string
+	Model    json.RawMessage
+
+	Created time.Time
+	Updated time.Time
+
+	CreatedBy int64
+	UpdatedBy int64
+}
+
+var (
+	// errLibraryPanelAlreadyAdded is an error when you add a library panel that already exists.
+	errLibraryPanelAlreadyAdded = fmt.Errorf("library panel with that title already exists")
+)
+
+// Commands
+
+// addLibraryPanelCommand is the command for adding a LibraryPanel
+type addLibraryPanelCommand struct {
+	OrgId        int64                `json:"-"`
+	FolderId     int64                `json:"folderId"`
```

Comment:
[Initialisms are written with consistent case in Go](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)
```suggestion
	OrgID        int64                `json:"-"`
	FolderID     int64                `json:"folderId"`
```

Reply:
That way (if it's `OrgID` instead of `OrgId`) xorm by default will assume that the respective column is `org_i_d` instead of `org_id` therefore xorm tags should be added for alerting the default behaviour:
```suggestion
	OrgId        int64                `json:"-"` `xorm:"org_id"`
	FolderId     int64                `json:"folderId"` `xorm:"folder_id"`
```

Reply:
Thank you I'll look into this.

Reply:
@papagian This isn't a database model type though, right?

Reply:
Could you comment on the `LibraryPanel` type instead? It should apply there.

Reply:
Yes it is

Reply:
@aknuds1 is right is not used directly by xorm (`LibraryPanel` [is passed](https://github.com/grafana/grafana/blob/a16134544962de0cb7924185bad5fdd29c36efd7/pkg/services/librarypanels/database.go#L35) to Insert) but it's general good to keep this in mind since the xorm docs are not that clear about it.

Reply:
So when do you know if something is used directly by xorm? Newb question 🙈

Reply:
I'm not good at definitions but I'll risk saying if it's a struct mapping a database table.

Reply:
@hugohaggmark You can only tell by seeing if the type is used for xorm queries or insertions.

@papagian There's no point in adding xorm annotations to a type that isn't used as a xorm model (i.e., used for queries or insertions). I don't think xorm docs are needed to clarify here, since there's just no way for those fields to be interpreted by xorm.

Reply:
@aknuds1 yes, my bad I had just confused `LibraryPanel ` and `addLibraryPanelCommand `.

Reply:
@papagian Aha thanks for clarifying! I agree that xorm's docs aren't very clear about field naming directives.

Reply:
I've fixed this in my latest commit (I hope)